src/components/MainTable/MainTable.tsx

```diff
@@ -1,15 +1,58 @@
 import PropTypes from "prop-types";
 import React, { useEffect, useState } from "react";
+import type { ReactNode } from "react";
+import type { SortDirection } from "types";
 
 import Pagination from "../Pagination";
 import Table from "../Table";
+import type { TableProps } from "../Table";
 import TableRow from "../TableRow";
 import TableHeader from "../TableHeader";
 import TableCell from "../TableCell";
 
 import "./MainTable.scss";
+import { TSFixMe } from "index";
 
-const updateSort = (setSortKey, setSortDirection, sortKey, sortDirection) => {
+type Header = {
```

Comment:
Could this type be imported from the TableHeader component?

Reply:
again, differing types:

```
type Header = {
  content?: ReactNode;
  className?: string;
  sortKey?: string;
};
```

```
// TableHeader.tsx

type Props = {
  children?: ReactNode;
  className?: string;
  sort?: SortDirection;
  onClick?: (evt: MouseEvent) => void;
} 
```

Reply:
Yeah maybe this should intersect using an Omit or just intersect `} & HTMLAttributes<HTMLTableHeaderCellElement>;`